### ISSUE TITLE
use python3 as default

### ISSUE THREAD

**Issue by hxin** [sidbdri/cookiecutter-de_analysis_skeleton#99]
move to use python3 by default.

This may involve change settings in docker image as well as sidb server, so by default, python3 VE will be created instead of python2.

This requires some testing such as the get_gene_length function, making sure that works under py3.

**Comment by hxin** [comment]
This is now implemented in docker from image version 1.4.6 onwards.

https://github.com/sidbdri/Docker/blob/4feb9bff64ff77d98dca60f4c08a2bc3d7cf1be5/CHANGELOG.md

sidb remains using py2 for now.

**Comment by hxin** [comment]
So for now we decide to keep the system _python_ command pointing to python2 as it was.
We forced the project script to create python3 VE in e39bf2adacdc334cdde6538b7fc52260b6c88d2b

This will create some problem as the transcript_utils, sidbdri_utils and sargasso were developed under python2 with some dependency version fixed, which will not work under python3.

I have tested the dependency versions by removing all the fixed versions and letting pip3 to decide which version to install, and this results in a set of versions that works under python3. The test results are logged [here](https://github.com/sidbdri/Docker/issues/30).

For now, we created a branch for each of the above repo removing all the fixed-version from the dependency packages.

The project script will install all those dependencies base on the versions tested [here](https://github.com/sidbdri/Docker/issues/30), and use the version-less transcript_utils, sidbdri_utils and sargasso branches.
